lib/spack/spack/test/build_environment.py

```diff
@@ -0,0 +1,85 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+import unittest
+
+import spack
+from llnl.util.filesystem import join_path
+from spack.build_environment import dso_suffix, _static_to_shared_library
+from spack.util.executable import Executable
+
+
+class BuildEnvironmentTest(unittest.TestCase):
+
+    def setUp(self):
+        self.cc = Executable(join_path(spack.build_env_path, "cc"))
+        self.ld = Executable(join_path(spack.build_env_path, "ld"))
+        self.cpp = Executable(join_path(spack.build_env_path, "cpp"))
+        self.cxx = Executable(join_path(spack.build_env_path, "c++"))
+        self.fc = Executable(join_path(spack.build_env_path, "fc"))
+
+        self.realcc = "/bin/mycc"
+        self.prefix = "/spack-test-prefix"
+
+        os.environ['SPACK_CC'] = self.realcc
+        os.environ['SPACK_CXX'] = self.realcc
+        os.environ['SPACK_FC'] = self.realcc
+
+        os.environ['SPACK_PREFIX'] = self.prefix
+        os.environ['SPACK_ENV_PATH'] = "test"
+        os.environ['SPACK_DEBUG_LOG_DIR'] = "."
+        os.environ['SPACK_COMPILER_SPEC'] = "gcc@4.4.7"
+        os.environ['SPACK_SHORT_SPEC'] = "foo@1.2"
+
+        os.environ['SPACK_CC_RPATH_ARG']  = "-Wl,-rpath,"
+        os.environ['SPACK_CXX_RPATH_ARG'] = "-Wl,-rpath,"
+        os.environ['SPACK_F77_RPATH_ARG'] = "-Wl,-rpath,"
+        os.environ['SPACK_FC_RPATH_ARG']  = "-Wl,-rpath,"
+
+        if 'SPACK_DEPENDENCIES' in os.environ:
+            del os.environ['SPACK_DEPENDENCIES']
+
+    def test_static_to_shared_library(self):
```

Comment:
This test is fine but can be moved to module scope if it's a pytest test.  It should take the `build_environment` fixture as a parameter.